src/status.f

```diff
@@ -81,8 +72,17 @@ RECURSIVE SUBROUTINE STATUS(LUNIT,LUN,IL,IM)
 C  SEE IF UNIT IS ALREADY CONNECTED TO BUFR ARCHIVE LIBRARY SOFTWARE
 C  -----------------------------------------------------------------
 
+      IF ( .NOT. ALLOCATED(IOLUN) ) THEN
```

Comment:
I think this should call bort().

I don't see how the user can proceed with any function which tries to call status() before initialization of the internal arrays. So even if status() issues a warning and returns, there will still be a segfault when the function calling status tries to use internal memory.